internal/consumer/http.go

```diff
@@ -0,0 +1,61 @@
+package consumer
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+type DataService interface {
+	getTopStories() ([]int, error)
+	getItem(id int) (*Item, error)
+}
+
+type HttpService struct {}
+
+const hnUrl string = "https://hacker-news.firebaseio.com/"
+
+func (hs HttpService) getTopStories() ([]int, error) {
+	url := fmt.Sprintf("%sv0/topstories.json?print=pretty", hnUrl)
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, errors.Wrap(err, "Get top stories: ")
+	}
+	responseData, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, errors.Wrap(err, "Get top stories: ")
+	}
+
+	var ids = []int{}
+	err = json.Unmarshal(responseData, &ids)
+	if err != nil {
+		return nil, errors.Wrap(err, "Get top stories: ")
+	}
+
+	return ids, nil
+}
+
+func (hs HttpService) getItem(id int) (*Item, error) {
+	url := fmt.Sprintf("%sv0/item/%d.json?print=pretty", hnUrl, id)
+	resp, err := http.Get(url)
```

Comment:
I would consider adding the http client onto your HTTPService struct which might help in testing